### ISSUE TITLE
Not able to give separate permissions to queries that call lambdas

### ISSUE THREAD

**Issue by mguja-fcc** [aws-amplify/amplify-cli#8605]
Not able to give separate permissions to queries that call lambdas. For example, I have two queries and want to give separate permissions to. Query `invokeLambda`  any logged in user should get access to but access to `invokeGetGroups` should be access only by Admins group. How can I accomplish this ?

```type Query {
  invokeLambda: String @function(name: "pythonLambda-${env}") 
  invokeGetGroups: String @function(name: "getGroups-${env}")
  @auth(
      rules: [
        { allow: groups, groups: ["Admins"] }
        ])
}
```

Amplify CLI version: 6.2.1

**Comment by GeorgeBellTMH** [comment]
I agree it would be great if @auth could be used with functions in graphql...but you can also just add the logic into the lambda itself until they develop things a bit more...for example: event.identity.claims["cognito:groups"].includes("admin")
